zynqtgui/sketchpad/sketchpad_clip.py

```diff
@@ -577,106 +1012,35 @@ def set_path(self, path, should_copy=True):
-            self.audioSource.setGrainScan(float(self.__get_metadata_prop__("ZYNTHBOX_GRAINERATOR_SCAN", 0)))
-            self.audioSource.setGrainInterval(float(self.__get_metadata_prop__("ZYNTHBOX_GRAINERATOR_INTERVAL", 10)))
-            self.audioSource.setGrainIntervalAdditional(float(self.__get_metadata_prop__("ZYNTHBOX_GRAINERATOR_INTERVAL_ADDITIONAL", 10)))
-            self.audioSource.setGrainSize(float(self.__get_metadata_prop__("ZYNTHBOX_GRAINERATOR_SIZE", 100)))
-            self.audioSource.setGrainSizeAdditional(float(self.__get_metadata_prop__("ZYNTHBOX_GRAINERATOR_SIZE_ADDITIONAL", 50)))
-            self.audioSource.setGrainPanMinimum(float(self.__get_metadata_prop__("ZYNTHBOX_GRAINERATOR_PAN_MINIMUM", -1)))
-            self.audioSource.setGrainPanMaximum(float(self.__get_metadata_prop__("ZYNTHBOX_GRAINERATOR_PAN_MAXIMUM", 1)))
-            self.audioSource.setGrainSustain(float(self.__get_metadata_prop__("ZYNTHBOX_GRAINERATOR_SUSTAIN", 0.3)))
-            self.audioSource.setGrainTilt(float(self.__get_metadata_prop__("ZYNTHBOX_GRAINERATOR_TILT", 0.5)))
-            self.audioSource.setGrainPitchMinimum1(float(self.__get_metadata_prop__("ZYNTHBOX_GRAINERATOR_PITCH_MINIMUM1", 1.0)))
-            self.audioSource.setGrainPitchMaximum1(float(self.__get_metadata_prop__("ZYNTHBOX_GRAINERATOR_PITCH_MAXIMUM1", 1.0)))
-            self.audioSource.setGrainPitchMinimum2(float(self.__get_metadata_prop__("ZYNTHBOX_GRAINERATOR_PITCH_MINIMUM2", 1.0)))
-            self.audioSource.setGrainPitchMaximum2(float(self.__get_metadata_prop__("ZYNTHBOX_GRAINERATOR_PITCH_MAXIMUM2", 1.0)))
-            self.audioSource.setGrainPitchPriority(float(self.__get_metadata_prop__("ZYNTHBOX_GRAINERATOR_PITCH_PRIORITY", 0.5)))
-            self.audioSource.progressChanged.connect(self.progress_changed_cb, Qt.QueuedConnection)
-            self.audioSource.gainAbsoluteChanged.connect(self.updateGain, Qt.QueuedConnection)
-            self.audioSource.playbackStyleChanged.connect(self.saveMetadata, Qt.QueuedConnection)
-
-        self.reset_beat_count()
-        self.set_length(self.__length__, True)
-        self.set_start_position(self.__start_position__, True)
-        self.set_loop_delta(self.__loop_delta__, True)
-        self.set_speedRatio(self.__speed_ratio__, True)
-        self.set_pitch(self.__pitch__, True)
-        self.set_gain(self.__gain__, True)
-        self.set_snap_length_to_beat(self.__snap_length_to_beat__, True)
-        self.update_synced_values()
 
         self.path_changed.emit()
-        self.__read_metadata__()
         self.duration_changed.emit()
         self.is_playing_changed.emit()
         if self.is_channel_sample:
             self.__song__.channelsModel.getChannel(self.row).samples_changed.emit()
         self.__song__.schedule_save()
 
+        # Schedule write as it should ensure metadata are writen for all loaded sketches
+        self.metadata.scheduleWrite()
```

Comment:
Are you sure we want to do this? It seems like if the defaults are fine, then we don't really need to do any saving until those default values have actually been changed... or am i missing something?

Reply:
Initially I thought this was required when loading a random sample and does not have any metadata yet and hence needs an explicit save to write the default values.

But thinking on it again, I think we can do away without explicitly writing after reading. If any metadata is changed it would eventually write all. And if nothing is changed then next time it loads, the default value would be set to the properties so that's fine again.